test-network-nano-bash/README.md

```diff
@@ -0,0 +1,83 @@
+![Terminal setup](terminal_setup.png)
+
+The following instructions will have you run simple bash scripts that set environment variable overrides for a component and then runs the component.
+The scripts contain only simple single-line commands so that they are easy to read and understand.
+If you have trouble running bash scripts in your environment, you can just as easily copy and paste the individual commands from the script files instead of running the script files.
+
+- cd to the `test-network-nano-bash` directory in each terminal window
+- In the first orderer terminal, run `./generate_artifacts.sh` to generate crypto material (calls cryptogen) and system and application channel genesis block and configuration transactions (calls configtxgen). The artifacts will be created in the `crypto-config` and `channel-artifacts` directories.
+- In the three orderer terminals, run `./orderer1.sh`, `./orderer2.sh`, `./orderer3.sh` respectively
+- In the four peer terminals, run `./peer1.sh`, `./peer2.sh`, `./peer3.sh`, `./peer4.sh` respectively
+- Note that each orderer and peer write their data (including their ledgers) to their own subdirectory under the `data` directory
+- In the four peer admin terminals, run `. peer1admin.sh`, `. peer2admin.sh`, `. peer3admin.sh`, `. peer4admin.sh` respectively
+
+Note the syntax of running the scripts. The peer admin scripts run with the dot command in order to source the script files in the respective shells. This is important so that the exported environment variables can be utilized by any subsequent user commands.
+
+The `peer1admin.sh` script sets the peer1 admin environment variables, creates the application channel `mychannel`, updates the channel configuration for the org1 gossip anchor peer, and joins peer1 to `mychannel`.
+The remaining peer admin scripts join their respective peers to `mychannel`.
+
+# Instructions for deploying and running the basic asset transfer sample chaincode
+
+Package and install the chaincode on peer1:
+
+```
+peer lifecycle chaincode package basic.tar.gz --path ../asset-transfer-basic/chaincode-go --lang golang --label basic_1
+
+peer lifecycle chaincode install basic.tar.gz
+```
+
+The chaincode install may take a minute since the `fabric-ccenv` chaincode builder docker image will be downloaded if not already available on your machine.
+Copy the returned chaincode package ID into an environment variable for use in subsequent commands (your ID may be different):
+
+```
+CC_PACKAGE_ID=basic_1:faaa38f2fc913c8344986a7d1617d21f6c97bc8d85ee0a489c90020cd57af4a5
+```
+
+Approve and commit the chaincode (only a single approver is required based on the lifecycle endorsement policy of any organization):
+
+```
+peer lifecycle chaincode approveformyorg -o orderer.example.com:6050 --channelID mychannel --name basic --version 1 --package-id $CC_PACKAGE_ID --sequence 1 --tls --cafile ${PWD}/crypto-config/ordererOrganizations/example.com/orderers/orderer.example.com/tls/ca.crt
```

Comment:
need to be 127.0.0.1:6050 and have `--ordererTLSHostnameOverride orderer.example.com`

Reply:
I updated to use 127.0.0.1:6050, don't need --ordererTLSHostnameOverride since I generated the crypto with SANS 127.0.0.1.